src/features/messages/components/Compose/index.tsx

```diff
@@ -0,0 +1,259 @@
+import React, { useState } from 'react'
+import { connect } from 'react-redux'
+import 'emoji-mart/css/emoji-mart.css'
+import { Picker } from 'emoji-mart'
+import { ReactMic } from 'react-mic'
+
+import { createMessage } from '../../MessagesStore'
+import ToolbarIconButton from '../ToolbarButton'
+import './Compose.css'
+
+interface IComposeProps {
+  createMessage: (msg: string, image: string[], audio: string) => void
+  isLoading?: boolean
+}
+
+interface IEmoji {
+  colons: string
+  emoticons: []
+  id: string
+  name: string
+  native: string
+  short_names: []
+  skin: null
+  unified: string
+}
+
+interface IAudio {
+  blob: Blob
+  startTime: number
+  stopTime: number
+  option: {
+    audioBitsPerSecond: number
+    mimeType: string
+  }
+  blobURL: string
+}
+
+function InputCompose(props: IComposeProps): JSX.Element {
+  const { createMessage, isLoading } = props
+  const [messageTyping, setMessage] = useState('')
+  const [messageImage, setMessageImage] = useState<string[]>([])
+  const [showEmoji, setShowEmoji] = useState(false)
+  const [showRecord, setShowRecord] = useState(false)
+  const [record, setRecord] = useState(false)
+  const [audio, setAudio] = useState('')
+
+  const onSend = () => {
+    createMessage(messageTyping, messageImage, audio)
+    setTimeout(scrollToBottom, 100)
+  }
+
+  const scrollToBottom = () => {
+    const msgList = document.getElementById('wrapper-list-message')
+    if (msgList) {
+      msgList.scrollTo({
+        left: 0,
+        top: msgList.scrollHeight,
+        behavior: 'smooth',
+      })
+    }
+  }
+
+  const handleSendMsg = () => {
+    if (messageTyping || messageImage || audio) {
+      onSend()
+      setMessage('')
+      setMessageImage([])
+      setAudio('')
+      setShowRecord(false)
+    }
+  }
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      handleSendMsg()
+    }
+  }
+
+  const onChangeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setMessage(e.target.value)
+  }
+
+  const onChangeEmoji = (emoji: IEmoji) => {
+    setMessage(_message => {
+      return _message + emoji.native
+    })
+  }
+
+  const onChangeShowEmoji = () => {
+    setShowEmoji(!showEmoji)
+  }
+
+  const onChangeShowImage = () => {
+    const element = document.getElementById('input-compose-image')
+    element?.click()
+  }
+
+  const onChangeImageCompose = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const files = e.target.files
+    if (files) {
+      Array.from(files).forEach((file: File) => {
+        const ext = file.name.split('.').pop()?.toLowerCase() || ''
+        if (!['jpg', 'png', 'jpeg'].includes(ext)) {
+          console.log('File invalid')
```

Comment:
We should do something better than a `console.log`. Perhaps show an error to a user?